<!-- -*- mode: XML -*- -->
<launch>
	<!-- necessary tf for robot_pose_ekf to connect to rest of robot -->
	<node machine="pri" pkg="tf" type="static_transform_publisher" name="base_footprint_tfB" 
		args="0 0 0.03 0 0 0  /base_footprint /base_link 25" />
	<node machine="pri" pkg="tf" type="static_transform_publisher" name="static_laser_tfB" 
		args="0.21 0 0.32 0 0 0  /base_link /static_laser 25" />

	<include file="$(find techx_bot)/launch/eaglemotor_params.launch" />
	<param name="odom_pub_rate" value="13"/>

  <!-- The odometry estimator -->
	<node machine="pri" pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
		<param name="freq" value="20.0"/>
		<param name="output_frame" value="odom"/>
		<param name="sensor_timeout" value="0.5"/>
		<param name="odom_used" value="true"/>
		<param name="imu_used" value="false"/>
		<param name="vo_used" value="true"/>

		<!--remap from="imu_data" to="/imu/data"/-->
		<remap from="vo" to="/scan_odom"/>
	</node>

</launch>
